tsl/test/sql/read_only.sql

```diff
@@ -12,8 +12,19 @@
 --
 CREATE TABLE test_table(time bigint NOT NULL, device int);
 
+-- EXPLAIN should work in read-only mode, when enabling in transaction.
+START TRANSACTION;
+SET transaction_read_only TO on;
+EXPLAIN (COSTS OFF) SELECT * FROM test_table;
```

Comment:
What happens when you do EXPLAIN ANALYZE ? 

What happens if the query performs a mutation? E.g., EXPLAIN ANALYZE select drop_chunks() or similar?

Reply:
These are checked inside the execution proper and will call `PreventIfReadOnly` there instead. This flag just disables "our" check (which I am not sure why it is there, since executing the statement should prevent that and we should not need to do this inside `process_utilities`).

Added tests to demonstrate that this is the case.